test/package/common.py

```diff
@@ -11,6 +12,10 @@ def __init__(self, *args, **kwargs):
         self._temporary_files = []
 
     def temp(self):
+        # Use a buffer instead of a temporary file in fbcode.
```

Comment:
I'm not sure if we want to have the behavior be different between fbcode/external. The benefit is that we get both paths tested well, the disadvantage is that all test code needs to handle both cases, and it makes the test logic slightly more confusing (not obvious that behavior is different when debugging an individual test).

What about: use `BytesIO` everywhere as convention, but have the tests in `test_save_load`  explicitly test the file-based path and disable those tests in sandcastle? It's more work, but I think it's good to have the tests be very explicit. Could be convinced either way though.

Reply:
I made this change and pushed again. The only thing I don't like about it is that people writing tests have to remember to `buffer.seek(0)` all the time.

What about introducing a context manager that automatically rewinds the test buffer that its `__enter__` returns:
```
with PackageTestBuffer() as buffer:
    # use buffer in exporter

# buffer is automatically rewound
# use buffer in importer
```

It is not intuitive that buffer can be used outside of the with statement but Python doesn't have scope anyway...

Reply:
Dunno if that's much better, you have to remember to do something special either way. At least `io` is a standard thing.